synedrion/src/cggmp21/params.rs

```diff
@@ -223,7 +246,12 @@ pub(crate) fn secret_uint_from_scalar<P: SchemeParams>(
     let scalar_len = scalar_bytes.expose_secret().len();
 
     debug_assert!(uint_len >= scalar_len);
-    repr.expose_secret_mut().as_mut()[uint_len - scalar_len..].copy_from_slice(scalar_bytes.expose_secret());
+    repr.expose_secret_mut()
+        .as_mut()
+        .get_mut(uint_len - scalar_len..)
+        // TODO(dp): @reviewers Do we need a better proof that this is true? If `Uint` is, say, 128 bits long…
```

Comment:
We do need some check for parameter consistency, but I'm not sure if it's possible to do in compile-time.

Reply:
Worth investigating perhaps, like have a `const fn validate_config() -> bool` that checks for configuration parameter consistency, e.g. that `<P::Paillier as PaillierParams>::Uint` is bigger than a `Point`.